src/app/plugins/controller.plugin.ts

```diff
@@ -13,20 +13,40 @@ export class PluginControl extends Plugin {
   }
 
   public validate(message: IMessage, args: string[]) {
-    return (args && args.length >= 2 && this.commandPattern.test(args.join(' ')));
+    return this.commandPattern.test(args.join(' '));
   }
 
 
   public async execute(message: IMessage, args: string[]): Promise<void> {
     const [method, pluginName] = args;
 
+    let plugin;
     try {
-      this.container.pluginService.setPluginState(pluginName, method === 'activate');
+      plugin = this.container.pluginService.setPluginState(pluginName, method === 'activate');
     } catch(e) {
       await message.channel.send(e.message);
       return;
     }
 
+    // Save data in persistently.
+    const pluginStateData = (await this.container.storageService.getCollections()).pluginState;
+    if (!pluginStateData) {
+      message.channel.send('Error connecting to the DB');
+      return;
+    }
+
+    try {
+      await pluginStateData
+        .updateOne({ name:  plugin.name }, 
```

Comment:
scope it by guild ID too, we have test servers on the same collection and scope by guildID